jmetal-core/src/main/java/org/uma/jmetal/util/experiment/component/EvolutionChart/HistoryData.java

```diff
@@ -0,0 +1,162 @@
+     */
+    public void printAllHistory(String baseDir, String indicator, String nameAlgorithm, String problemName){
+        String outputDir = baseDir + "/"+DEFAULT_BASE+"/"+nameAlgorithm+"/";
+        File f = new File(outputDir);
+        if(!f.exists()){
+            f.mkdirs();
+        }
+        try (OutputStream os = new FileOutputStream(outputDir+"data_"+indicator+"_"+nameAlgorithm+"_"+problemName+".dat")) {
+            PrintStream ps = new PrintStream(os);
+            for (int data = 0; data < numberOfData; data++) {
+                ps.print(data);
+                if(data<numberOfData-1){
+                    ps.print("\t");
+                }
+            }
+            ps.print("\n");
+            for (int test = 0; test < numberOfTest; test++) {
+                for (int data = 0; data < numberOfData; data++) {
+                    ps.print(history[data][test]);
+                    if(data<numberOfData-1){
+                        ps.print("\t");
+                    }
+                }
+                ps.print("\n");
+            }
+        } catch (IOException ex) {
+            Logger.getLogger(GenerateEvolutionChart.class.getName()).log(Level.SEVERE, null, ex);
+        }
+    }
+    
+    
+    public static Map<String, Double> calculateQualityIndicator(List<DoubleSolution> population,
+                                                                String problemName,
+                                                                Set<String> indicatorToSave) {
+        Front referenceFront;
+        Map<String, Double> indicators = new HashMap();
+        try {
+            referenceFront = new ArrayFront("/resources/pareto_fronts/" + problemName + ".pf");
+            FrontNormalizer frontNormalizer = new FrontNormalizer(referenceFront);
+            Front normalizedReferenceFront = frontNormalizer.normalize(referenceFront);
+            Front normalizedFront = frontNormalizer.normalize(new ArrayFront(population));
+            List<PointSolution> normalizedPopulation = FrontUtils.convertFrontToSolutionList(normalizedFront);
+            //List<String> ind = new ArrayList(indicatorToSave);
+            for (String string : indicatorToSave) {
+                if(string.equals("HV")){
+                    indicators.put("HV", new PISAHypervolume<PointSolution>(normalizedReferenceFront).evaluate(normalizedPopulation));
+                } else if(string.equals("Epsilon")){
+                    indicators.put("Epsilon", new Epsilon<PointSolution>(normalizedReferenceFront).evaluate(normalizedPopulation));
+                } else if(string.equals("IGD")){
+                    indicators.put("IGD", new InvertedGenerationalDistancePlus<PointSolution>(normalizedReferenceFront).evaluate(normalizedPopulation));
+                } else if(string.equals("Spread")){
+                    indicators.put("Spread", new Spread<PointSolution>(normalizedReferenceFront).evaluate(normalizedPopulation));
+                }
```

Comment:
Rather than strings, an `enum` seems relevant to me here. Moreover, that suite of conditions would be better factored as a switch-method, like:
```java
private static enum Indicator {HV, Epsilon, IGD, Spread}

private Double getIndicatorValue(Indicator indicator) {
	switch indicator {
		case HV: return new PISAHypervolume<PointSolution>(normalizedReferenceFront).evaluate(normalizedPopulation);
		case Epsilon: return new Epsilon<PointSolution>(normalizedReferenceFront).evaluate(normalizedPopulation);
		case IGD: return new InvertedGenerationalDistancePlus<PointSolution>(normalizedReferenceFront).evaluate(normalizedPopulation);
		case Spread: return new Spread<PointSolution>(normalizedReferenceFront).evaluate(normalizedPopulation);
		default: throw new IllegalArgumentException("Unmanaged indicator: " + indicator);
	}
}
```
It would reduce the loop to:
```java
for (String indicator : indicatorToSave) {
	indicators.put(indicator, getIndicatorValue(indicator));
}
```
Which can further be refined through lambdas (just for curiosity, there is no real added value here):
```java
indicatorToSave.stream().peek(indicator -> indicators.put(indicator, getIndicatorValue(indicator)))
```